src/prx/rinex_nav/evaluate.py

```diff
@@ -591,14 +598,16 @@ def find_ephemeris_index(row, df):
     df = df.reset_index(drop=True)
     df["ephemeris_index"] = df.index
 
-    query[query.ephemeris_index.isna()]["ephemeris_index"] = df.index[len(df) - 1]
+    # Use .loc to set the values in the query DataFrame to solve SettingWithCopyWarning
+    query.loc[query.ephemeris_index.isna(), "ephemeris_index"] = df.index[len(df) - 1]
     # Copy ephemerides into query dataframe
     # We are doing it this way around because the same satellite might show up multiple times in the query dataframe,
     # e.g. with different query times
     query = query.merge(df.drop(columns=["sv"]), on="ephemeris_index")
     # For Galileo satellites we can have both F/NAV and I/NAV ephemerides for the same satellite and time, keep
     # only one
     # Compute times w.r.t. orbit and clock reference times used by downstream computations
+    query = query[query["ephemeris_reference_time_isagpst"].notna() & (query["ephemeris_reference_time_isagpst"] != '')]
```

Comment:
It looks like the `ephemeris_reference_time_isagpst` column's `dtype` is string here, it should be `float`.